File: codewars/high-score.py
class HighScoreTable:
    
    def __init__(self, count):
        self.scores = []
        self.count = count

    def update(self, value): 
        if (len(self.scores) < self.count):
            self.scores.append(value)
        else:
            for i in range(len(self.scores) - 1, -1, -1):
                if (self.scores[i] < value):
                    self.scores[i] = value

        self.scores.sort(reverse=True)

    def reset(self):
        self.scores = []


highScoreTable = HighScoreTable(3)
print(highScoreTable.scores) # evaluates to True

highScoreTable.update(10)

print(highScoreTable.scores)

highScoreTable.update(8)
highScoreTable.update(12)
highScoreTable.update(5)
highScoreTable.update(10)

print(highScoreTable.scores)

highScoreTable.reset()

print(highScoreTable.scores)

File: codewars/high-score.js
class HighScoreTable {
    constructor(count) {
        this.count = count;
        this.scores = [];
    }

    update(value) {
        if (this.scores.length < this.count) {
            this.scores.push(value);
        } else {
            for (let i = this.scores.length - 1; i >= 0; i--) {
                if (this.scores[i] < value) {
                    this.scores[i] = value;
                }
            }

            this.scores.sort((a, b) => b - a);
        }
    }

    reset() {
        this.scores = [];
    }
}

let highScoreTable = new HighScoreTable(3)
console.log(highScoreTable.scores) // evaluates to True

highScoreTable.update(10)

console.log(highScoreTable.scores)

highScoreTable.update(8)
highScoreTable.update(12)
highScoreTable.update(5)
highScoreTable.update(10)

console.log(highScoreTable.scores)

highScoreTable.reset()

console.log(highScoreTable.scores)